docstring_parser/rest.py

```diff
@@ -139,11 +141,25 @@ def parse(text: str) -> Docstring:
             ) from ex
         args = args_chunk.split()
         desc = desc_chunk.strip()
+
         if "\n" in desc:
             first_line, rest = desc.split("\n", 1)
             desc = first_line + "\n" + inspect.cleandoc(rest)
 
-        ret.meta.append(_build_meta(args, desc))
+        # Add special handling for :type a: typename
+        if len(args) == 2 and args[0] == "type":
+            types[args[1]] = desc
+        elif len(args) in [1, 2] and args[0] == "rtype":
+            rtypes[None if len(args) == 1 else args[1]] = desc
+        else:
+            ret.meta.append(_build_meta(args, desc))
+    for meta in ret.meta:
```

Comment:
Let's add a blank line before the for loop for clarity that this is postprocessing

Reply:
Fixed.